tests/Aspire.Workload.Tests/PerTestFrameworkTemplatesTests.cs

```diff
@@ -0,0 +1,179 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Text;
+using System.Text.RegularExpressions;
+using System.Xml;
+using Microsoft.Playwright;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace Aspire.Workload.Tests;
+
+public abstract partial class PerTestFrameworkTemplatesTests : WorkloadTestsBase
+{
+    private static readonly XmlWriterSettings s_xmlWriterSettings = new() { ConformanceLevel = ConformanceLevel.Fragment };
+
+    [GeneratedRegex(@"^\s*//")]
+    private static partial Regex CommentLineRegex();
+
+    // Regex is from src/Aspire.Hosting.AppHost/build/Aspire.Hosting.AppHost.targets - _GeneratedClassNameFixupRegex
+    [GeneratedRegex(@"(((?<=\.)|^)(?=\d)|\W)")]
+    private static partial Regex GeneratedClassNameFixupRegex();
+
+    private readonly string _testTemplateName;
+
+    public PerTestFrameworkTemplatesTests(string testType, ITestOutputHelper testOutput) : base(testOutput)
+    {
+        _testTemplateName = testType;
+    }
+
+    public static TheoryData<string> ProjectNamesWithTestTemplate_TestData()
+    {
+        var data = new TheoryData<string>();
+        foreach (var name in GetProjectNamesForTest())
+        {
+            data.Add(name);
+        }
+        return data;
+    }
```

Comment:
This could be simplified:

```suggestion
    public static IEnumerable<object[]> ProjectNamesWithTestTemplate_TestData()
    {
        foreach (var name in GetProjectNamesForTest())
        {
            yield return [name];
        }
    }
```